applications/ShallowWaterApplication/python_scripts/coupling/depth_integration_input_process.py

```diff
@@ -166,22 +175,26 @@ def _MapToBoundaryCondition(self):
 
 
     def _CreateMapper(self):
-        mapper_settings = KM.Parameters("""{
-            "mapper_type": "nearest_neighbor",
-            "echo_level" : 0,
-            "search_settings" : {
-                "search_radius" : 0.0
-            }
-        }""")
-        min_point = KM.Point([ 1e6,  1e6,  1e6])
-        max_point = KM.Point([-1e6, -1e6, -1e6])
-        for node in self.interface_model_part.Nodes:
-            for i in range(3):
-                point = KM.Point(node)
-                min_point[i] = min([min_point[i], point[i]])
-                max_point[i] = max([max_point[i], point[i]])
-        distance = 1.05 * (max_point - min_point).norm_2()
-        mapper_settings["search_settings"]["search_radius"].SetDouble(distance)
+        domain_size = self.input_model_part.ProcessInfo[KM.DOMAIN_SIZE]
+        if domain_size == 2:
+            mapper_settings = KM.Parameters("""{
+                "mapper_type": "nearest_neighbor",
+                "echo_level" : 0,
+                "search_settings" : {
+                    "search_radius" : 0.0
+                }
+            }""")
+            min_point, max_point = KM.BoundingBox(self.interface_model_part.Nodes).GetPoints()
+            search_radius = 1.05 * (max_point - min_point).norm_2()
+            mapper_settings.AddEmptyValue("search_settings")
+            mapper_settings["search_settings"]["search_radius"].SetDouble(search_radius)
```

Comment:
is this faster?
this seems larger than what I use by default so my guess is that it is slower

But ofc this is only a guess :)

Reply:
Here I have a very specific configuration. In the 2D case I need to map from one node to all the BC, so I need a big search radius. The 3D case is more standard.